File: longestConsecutive.js
var longestConsecutive = function (nums) {
  let numSet = new Set(nums);

  let max = 0;

  for (let num of numSet) {
    if (!numSet.has(num - 1)) {
      let count = 0;
      while (numSet.has(num)) {
        count++;
        num++;
        max = Math.max(max, count);
      }
    }
  }

  return max;
};

// const nums = [100,4,200,1,3,2]
const nums = [0, 3, 7, 2, 5, 8, 4, 6, 0, 1];
// const nums = [0,0]

console.log(longestConsecutive(nums));

File: longestConsecutive.py
from typing import List

class Solution:
  def longestConsecutive(self, nums: List[int]) -> None:

     num_set = set(nums)

     max_num = 0

     for num in num_set:
       if num - 1 not in num_set:
         cur_num = num
         count = 1
         while cur_num + 1 in num_set:
           cur_num += 1
           count += 1
         max_num = max(max_num, count)
     return max_num